ui.apps/src/main/content/jcr_root/apps/core/cif/clientlibs/common/js/CommerceGraphqlApi.js

```diff
@@ -11,175 +11,183 @@
  *     governing permissions and limitations under the License.
  *
  ******************************************************************************/
+'use strict';
+
+class CommerceGraphqlApi {
+    constructor(props) {
+        if (!props || !props.endpoint) {
+            throw new Error(
+                'The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object'
+            );
+        }
 
-(function() {
-    'use strict';
-    //todo update the proxy rules so that this gets router through the dispatcher
-    const imageUrlPrefix = '/magento/img';
-
-    class CommerceGraphqlApi {
-        constructor(props) {
-            if (!props.endpoint) {
-                throw new Error(
-                    'The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object'
-                );
-            }
+        this.endpoint = props.endpoint;
+    }
 
-            this.endpoint = props.endpoint;
+    async _fetch(url, params) {
+        let response = await fetch(url, params);
+        if (!response.ok || response.errors) {
+            let message = await response.text();
+            throw new Error(message);
         }
+        return response.json();
+    }
 
-        async _fetch(url, params) {
-            let response = await fetch(url, params);
-            if (!response.ok || response.errors) {
-                let message = await response.text();
-                throw new Error(message);
+    async _fetchGraphql(query, cached) {
```

Comment:
This with later calling it by `_fetchGraphql(query, true);` will always result in a `GET` request. Wouldn't it be worth to make this a config property as well, same like `endpoint`. WDYT?

Reply:
I'm not comfortable with having the request method controlled by a different argument (in this case, `cached`). I'd introduced caching by default (i.e. have an `ignoreCache` attribute) and have a third optional parameter to control the method.

AFAIU we don't get the cached results with POST requests, but I think this is a Magento thing right?

Reply:
It depends on the query if it should be sent in a cache-able query or not. For example use specific queries (like cart) or mutations must be sent as non-cacheable POST query.

For Magento, only GET requests are cached and POST requests are always un-cached. So if we have a global parameter, how do we differentiate between cache-able and non-cache-able data?

Reply:
> that doesn't make sense to me tbh.

No, you're right it doesn't. 

We should still revert the condition, i.e. make the requests always cacheable and ignore the cache as needed. This is how I've seen it working with other clients as well (i.e. Apollo) - everything is cached by default and you have the option to bypass the cache if needed.

Reply:
@dplaton with caching by default (= `GET` by default) we set the minimum Magento version to 2.3.2.
@herzog31 we should add that to the readme/wiki that one can achieve 2.3.1 compatibility by changing to `POST` then.

Reply:
@mhaack So you're referring to an option to globally disable GraphQL `GET` requests to enable 2.3.1 compatibility?

Reply:
Additionally, maybe we should not call it `cached` at all just "request method" or something. Since `GET` requests are cached is Magento behavior and also on Magento side requires additional config, so not all `GET` responses might be cached in the end.

Reply:
> @herzog31 So you're referring to an option to globally disable GraphQL GET requests to enable 2.3.1 compatibility?

Not sure if globally is needed, just document that they are aware of the option.

Reply:
@mhaack I added a small section to the readme.